captum/attr/_utils/class_summarizer.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from captum.attr._utils.stat import Stat
+from captum.attr._utils.summarizer import Summarizer
+
+
+class ClassSummarizer:
+    """
+    Used to keep track of summaries for associated classes. The
+    classes/labels can be of any type that is supported by `dict`.
+
+    This also keeps track of an aggregate of all class summaries.
+    """
+
+    def __init__(self, stats: List[Stat]):
+        self.stats = stats
+        self.all_summary = Summarizer(stats=stats)
+        self.summaries: Dict[Any, Summarizer] = defaultdict(
+            lambda: Summarizer(stats=stats)
+        )
+
+    def update(
+        self,
+        x: Union[Tensor, Tuple[Tensor, ...]],
+        labels: Optional[Union[Any, List[Any]]] = None,
+    ):
+        """
+        Updates the stats of the summarizer, optionally associated to classes.
+
+        Args:
+            x (Tensor or Tuple[Tensor, ...]):
+                The input tensor to be summarised. The first
+                dimension of this input must be associated to
+                the amount of `labels` specified; unless there
+                is one labels represents one label (in that case
+                the first dimension can optionally be unsqueezed).
+            labels (Any, List[Any], optional):
+                The associated labels for `x`. If Any, we
+                assume x does not represent a batch of inputs.
+
+                If this is None we simply aggregate the total summary.
+        """
```

Comment:
nit: to avoid nested-ness, I'd probably do something like:
```
if labels is None:
    self.all_summary.update(x)
    return

# The rest of the code goes here
if not isinstance(x, tuple):

.... 

```